src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -25,6 +25,9 @@ public class ViewBuffer : IHtmlContentBuilder
         private readonly IViewBufferScope _bufferScope;
         private readonly string _name;
         private readonly int _pageSize;
+        private ViewBufferPage _singlePage = null;          // Limits allocation if the ViewBuffer has only one page (frequent case).
+        private List<ViewBufferPage> _multiplePages = null; // Allocated only if necessary
+        private ViewBufferPage _currentPage = null;
```

Comment:
I'm pretty neutral on this. It's one extra reference we need to ensure is kept up to date at all times, but it does make it a bit easier to reason about.